tools/flash.sh

```diff
@@ -65,5 +65,10 @@ if [ $tockload == "n" ]; then
 	exit 0
 fi
 
+if ! [ -x "$(command -v tockloader)" ]; then
+    echo "Skipping flashing as tockloader isn't installed"
+    exit 0
+fi
+
```

Comment:
Does the workflow depend on the condition you added (lines 68-72)? If so, then should we extract the "convert ELF into TAB" part of `flash.sh` into a separate script that is called by your workflow? If I understand this correctly it seems like a hack.

Reply:
I was thinking about this.

Currently running `make opentitan` will build all the examples as elf files, which makes sense. It does this by calling `cargo build`
Running `make flash-opentitan EXAMPLE=<app>` will build a elf and TBF of the specified example and for some targets try to flash it. It does this by running `cargo run`

This change means that if tockloader isn't installed (like in the CI) we don't try to flash the app. It doesn't really too hacky, but I sort of agree it is a little.

I did think about changing `make opentitan` to build elft **AND** TBFs by running `elf2tab` but that ends up being more complex then I was hoping and requires running scripts/commands in the Makefile which breaks the current "you can use Make or Cargo" idea that we currently have.

In the long run I think it would make sense for `make opentitan` to build both ELFs and TBFs, but it's not that simple so it probably should be separate.

Reply:
Okay, I see what you mean. I guess I'm fine with this as-is.